OpenRA.Game/FileSystem/FileSystem.cs

```diff
@@ -17,51 +17,21 @@
 
 namespace OpenRA.FileSystem
 {
-	public interface IFolder : IDisposable
+	public class FileSystem
 	{
-		Stream GetContent(string filename);
-		bool Exists(string filename);
-		IEnumerable<uint> ClassicHashes();
-		IEnumerable<uint> CrcHashes();
-		IEnumerable<string> AllFileNames();
-		void Write(Dictionary<string, byte[]> contents);
-		int Priority { get; }
-		string Name { get; }
-	}
-
-	public static class GlobalFileSystem
-	{
-		public static List<IFolder> MountedFolders = new List<IFolder>();
-		static Cache<uint, List<IFolder>> classicHashIndex = new Cache<uint, List<IFolder>>(_ => new List<IFolder>());
-		static Cache<uint, List<IFolder>> crcHashIndex = new Cache<uint, List<IFolder>>(_ => new List<IFolder>());
-
-		public static List<string> FolderPaths = new List<string>();
-
-		static void MountInner(IFolder folder)
-		{
-			MountedFolders.Add(folder);
-
-			foreach (var hash in folder.ClassicHashes())
-			{
-				var l = classicHashIndex[hash];
-				if (!l.Contains(folder))
-					l.Add(folder);
-			}
+		public readonly List<string> FolderPaths = new List<string>();
+		public readonly List<IFolder> MountedFolders = new List<IFolder>();
 
-			foreach (var hash in folder.CrcHashes())
-			{
-				var l = crcHashIndex[hash];
-				if (!l.Contains(folder))
-					l.Add(folder);
-			}
-		}
+		readonly Dictionary<string, Assembly> assemblyCache = new Dictionary<string, Assembly>();
 
-		static int order = 0;
+		int order;
+		Cache<uint, List<IFolder>> crcHashIndex = new Cache<uint, List<IFolder>>(_ => new List<IFolder>());
+		Cache<uint, List<IFolder>> classicHashIndex = new Cache<uint, List<IFolder>>(_ => new List<IFolder>());
 
-		public static IFolder CreatePackage(string filename, int order, Dictionary<string, byte[]> content)
+		public IFolder CreatePackage(string filename, int order, Dictionary<string, byte[]> content)
 		{
 			if (filename.EndsWith(".mix", StringComparison.InvariantCultureIgnoreCase))
-				return new MixFile(filename, order, content);
+				return new MixFile(this, filename, order, content);
 			if (filename.EndsWith(".zip", StringComparison.InvariantCultureIgnoreCase))
 				return new ZipFile(filename, order, content);
```

Comment:
Can you pass `this` to ZipFile too?  This will make life easier when we extract an interface for package types.